collections/src/main/scala/strawman/collection/immutable/Map.scala

```diff
@@ -84,6 +105,35 @@ trait MapOps[K, +V, +CC[X, +Y] <: MapOps[X, Y, CC, _], +C <: MapOps[K, V, CC, C]
 
 object Map extends MapFactory[Map] {
 
+  final class WithDefault[K, +V](underlying: Map[K, V], d: K => V) extends Map[K, V] {
+    // These factory methods will lose the default value
+    def iterableFactory = underlying.iterableFactory
+    def mapFactory: MapFactory[Map] = underlying.mapFactory
+    protected[this] def mapFromIterable[K2, V2](it: collection.Iterable[(K2, V2)]): Map[K2,V2] = mapFactory.from(it)
+
+    // Specific building will keep the default but may lose the precise underlying type because our own V can be
+    // a supertype of the underlying collection's V so we cannot rebuild with potentially new values that are not
+    // valid for the underlying collection.
+    protected[this] def fromSpecificIterable(coll: collection.Iterable[(K, V)]): Map[K,V] = new WithDefault[K, V](mapFactory.from(coll), d)
+    protected[this] def newSpecificBuilder(): mutable.Builder[(K, V), Map[K,V]] = {
+      val ub = mapFactory.newBuilder[K, V]()
+      new mutable.Builder[(K, V), Map[K, V]] {
+        def clear(): Unit = ub.clear()
+        def result(): Map[K, V] = new WithDefault[K, V](ub.result(), d)
+        def add(elem: (K, V)): this.type = { ub.add(elem); this }
+      }
+    }
```

Comment:
It seems that it should be possible to use `mapFactory.newBuilder[K, V]().mapResult(WithDefault[K, V](_, d))` instead.